hal/src/nRF52840/usart_hal.cpp

```diff
@@ -289,6 +289,7 @@ class Usart {
         enabled_ = false;
         transmitting_ = false;
         receiving_ = 0;
+        rxBuffer_.reset();
```

Comment:
Why? There might be some leftover data that may be needed.

Reply:
Without reseting the `rxBuffer_`, the receiver cannot work properly. It will fail the unit test. If the RX buffer needs to be retained, I'll try to figure out the root cause.

Reply:
I think the unit test failed due to not flushing the TX buffer as pointed out: https://github.com/particle-iot/device-os/pull/2125#discussion_r441307149. I'll add a test case to verify if the RX buffer is retained.